meiga/result.py

```diff
@@ -20,23 +30,23 @@ def __init__(
         self._value_failure = failure
         self._assert_values()
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         status = "failure"
         value = self.value.__repr__()
         if self._is_success:
             status = "success"
             value = self.value
         return f"Result[status: {status} | value: {value}]"
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
         if isinstance(other, Result):
             return (
                 self._value_success == other._value_success
                 and self.value == other.value
             )
         return False
 
-    def _assert_values(self):
+    def _assert_values(self) -> Union[NoReturn, None]:
```

Comment:
I kinda thought `NoReturn` is reserved only for those functions that never return at all, like infinite loops or guaranteed throws. Is it semantically correct to use it a part of `Union`? Because on that logic, every function that potentially throws can then be marked as `NoReturn`...